### PR TITLE
Discovery credentials 080521a

### PR DESCRIPTION
**Pull request by jbranson000**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -14,20 +14,22 @@ governing permissions and limitations under the License.
 const Swagger = require('swagger-client')
 const { codes } = require('./SDKErrors')
 const logger = require('@adobe/aio-lib-core-logging')('aio-lib-analytics', { level: process.env.LOG_LEVEL })
+const fetch = require('node-fetch')
```
Comment (shazron) [inline src/index.js:17]: I think you forgot to `npm install node-fetch`. If you did, you probably forgot to add the updated `package.json` in this PR

File: src/index.js [hunk 2]
```diff
@@ -47,15 +49,24 @@ function init (companyId, apiKey, token) {
...
-  async init (companyId, apiKey, token) {
+  async init (companyId, apiKey, token, options) {
+    // console.debug(companyId, apiKey, token, imsOrgId)
     const initErrors = []
-    if (!companyId) {
-      initErrors.push('companyId')
+    if (!companyId || companyId === null) {
+      const discoveryResponse = await this.getDiscoveryCredentials(apiKey, token)
+      for (const i in discoveryResponse.imsOrgs) {
+        if (discoveryResponse.imsOrgs[i].imsOrgId === options?.imsOrg) {
```
Comment (shazron) [inline src/index.js:64]: `options?.imsOrg` fails the CI. Optional chaining is only on node 14. We need to support both 10 and 12 still.

**Review by shazron — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (jbranson000) [reply src/index.js:17]: You are right.  I've resolved that in the latest PR.

File: src/index.js (hunk 2 above)
Comment (jbranson000) [reply src/index.js]: I've also resolved this in the latest push

File: src/index.js [hunk 3]
```diff
@@ -61,10 +61,14 @@ class AnalyticsCoreAPI {
     if (!companyId || companyId === null) {
       const discoveryResponse = await this.getDiscoveryCredentials(apiKey, token)
       for (const i in discoveryResponse.imsOrgs) {
-        if (discoveryResponse.imsOrgs[i].imsOrgId === options?.imsOrg) {
-          companyId = discoveryResponse.imsOrgs[i].companies[0].globalCompanyId
-          var companyName = discoveryResponse.imsOrgs[i].companies[0].companyName
-          break
+        if (options !== null || options !== undefined) {
+          if (options.imsOrg !== null || options.imsOrg !== undefined) {
+            if (discoveryResponse.imsOrgs[i].imsOrgId === options?.imsOrg) {
```
Comment (shazron) [inline src/index.js:66]: you still have optional chaining here

**Comment by shazron** [comment]
Two unit test failures. My suggestion is run `npm test` locally before committing so we don't have a long feedback cycle via the Github Actions CI.

**Pull request closed without merging**